lib/s3/managed_upload.js

```diff
@@ -173,10 +173,10 @@ AWS.S3.ManagedUpload = AWS.util.inherit({
           on('end', function() {
             self.isDoneChunking = true;
             self.numParts = self.totalPartNumbers;
+            self.fillQueue.call(self);
```

Comment:
Calling this without further data results in a noop, but when the stream has more data to read it increments the `self.numParts (self.totalPartNumbers )` https://github.com/aws/aws-sdk-js/blob/master/lib/s3/managed_upload.js#L449 causing the if below to be skipped.